org.bardsoftware.eclipsito/src/org/bardsoftware/impl/eclipsito/BootImpl.java

```diff
@@ -35,6 +39,33 @@ public void run(String application, File modulesDir, String descriptorPattern, L
       return plugins;
     }
 
+  private File getVersionDir(File modulesDir) {
+    File versionFile = new File(modulesDir.getPath() + File.separator + "VERSION");
+    String version = null;
+    try (BufferedReader br = new BufferedReader(new FileReader(versionFile))) {
+      version = br.readLine();
+    } catch (IOException e) {
+      Boot.LOG.severe("No version file found");
+    }
+
+    File additionalVersionFile = new File(System.getProperty("user.home") + File.separator + ".ganttproject.d" + File.separator + "plugins" + File.separator + "VERSION");
```

Comment:
In fact, we don't want to hardcode some specific folders here. 

We can pass a list of such folder in `-plugins-res` or `-plugins-dir` argument or in the corresponding attributes of config file. The list items can be separated with `File.pathSeparator`

This will make the code here simpler, because you will get a list of possible `modulesDirs`: `getVersionDir(List<File> moduleDirs)`. However, in `Boot` class you'll need to split the list and convert every entry to File instance by applying code at lines Boot:126-133 

One additional trick: getting user home in Java code is pretty easy and uniform across platforms while in the launchers it is always different. We'll probably want to pass `~` in the first path component to indicate that it is user.home, and it should be replaced here with the value of `user.home` system property.

Thus, the argument of -plugins-res or -plugins-dir list will basically look as follows:
`plugins:~/.ganttproject.d/plugins`

Reply:
If `/` is like a separator then I can use it as argument for `String#split` function. But If at first place will be ~ then we need to have additional logic how to understand that in result of split function (`~`, `.ganttproject.d`, `plugins`) the first value is related to the second to indicate that it is `user.home` directory
Maybe would be better to have another split value? Like `+` or `;` etc?